### PR TITLE
LibAudio: Port to `Core::Stream`

### PR DESCRIPTION
**Pull request by LucasChollet**
This will remove any usage of `Core::File` in LibAudio.

First, by deleting the plugin's method that return a `Core::File` object.
Both Wav and Flac plugins kept this only for compatibility.

Second, I ported the MP3 plugin that still used a `Core::File` object.

Finally, I factorized a common pattern, that arise from the common usage of `Core::Stream`, to the base class `LoaderPlugin`.

### CODE REVIEW

**Review by sin-ack — APPROVED** [review]
:yakkiss:

File: Userland/Libraries/LibAudio/FlacLoader.cpp [hunk 1]
```diff
@@ -26,33 +26,21 @@
...
-
-    auto maybe_stream = Core::Stream::BufferedFile::create(MUST(Core::Stream::File::open(path, Core::Stream::OpenMode::Read)), FLAC_BUFFER_SIZE);
-    if (maybe_stream.is_error())
-        m_error = LoaderError { "Can't open file stream" };
-    else
-        m_stream = maybe_stream.release_value();
 }
 
-FlacLoaderPlugin::FlacLoaderPlugin(Bytes& buffer)
+FlacLoaderPlugin::FlacLoaderPlugin(Bytes const& buffer)
```
Comment (linusg) [inline Userland/Libraries/LibAudio/FlacLoader.cpp:33]: Just pass it by value, no need to pass a span by const reference (same as StringView)

File: Userland/Libraries/LibAudio/FlacLoader.h [hunk 2]
```diff
@@ -115,6 +114,7 @@ class FlacLoaderPlugin : public LoaderPlugin {
     // keep track of the start of the data in the FLAC stream to seek back more easily
     u64 m_data_start_location { 0 };
     OwnPtr<Core::Stream::SeekableStream> m_stream;
+    Optional<Bytes const&> m_backing_memory;
```
Comment (linusg) [inline Userland/Libraries/LibAudio/FlacLoader.h]: Likewise, just `Optional<Bytes>`

File: Userland/Libraries/LibAudio/MP3Loader.cpp [hunk 3]
```diff
@@ -19,9 +19,18 @@ MP3LoaderPlugin::MP3LoaderPlugin(StringView path)
 {
 }
 
+MP3LoaderPlugin::MP3LoaderPlugin(Bytes const& buffer)
```
Comment (linusg) [inline Userland/Libraries/LibAudio/MP3Loader.cpp:22]: Likewise

File: Userland/Libraries/LibAudio/Loader.cpp [hunk 4]
```diff
@@ -11,6 +11,26 @@
 
 namespace Audio {
 
+LoaderPlugin::LoaderPlugin(StringView path)
+    : m_path(path)
+{
+}
+
+LoaderPlugin::LoaderPlugin(Bytes const& buffer)
```
Comment (linusg) [inline Userland/Libraries/LibAudio/Loader.cpp:19]: Likewise

**Review by linusg — APPROVED** [review]

**Pull request merged**